### PR TITLE
Update avro and nimbus dependencies

### PR DESCRIPTION
**Pull request by conniey**
* Updates avro dependencies from 1.9.2 to 1.10.1.
* Updates microsoft-eventhubs dependency for nimbus-jwt from 6.0.1 to 8.19

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -66,8 +66,8 @@ javax.validation:validation-api;2.0.1.Final
 mysql:mysql-connector-java;8.0.22
 net.minidev:json-smart;2.3
 org.apache.ant:ant;1.9.4
-org.apache.avro:avro;1.9.2
-org.apache.avro:avro-maven-plugin;1.9.2
+org.apache.avro:avro;1.10.1
```
Comment (alzimmermsft) [inline eng/versioning/external_dependencies.txt:69]: This should also resolve the CVE from Jackson being used as it was upgraded

**Review by alzimmermsft — APPROVED** [review]

**Review by srnagar — COMMENTED** [review]
Are we planning to release schema registry as well?

**Review by srnagar — APPROVED** [review]

**Comment by conniey** [comment]
> Are we planning to release schema registry as well?

I wanted to get an OK from @arerlend to ship it

**Review by arerlend — APPROVED** [review]
Lgtm. From EH perspective there would be no functional changes if we ship a new version, just avro version upgrade, so not mission-critical for us.

**Review by hemanttanwar — APPROVED** [review]

**Pull request merged**